network/p2p/network.go

```diff
@@ -227,17 +227,21 @@ func (n *Network) handleRegisterEngineRequest(parent irrecoverable.SignalerConte
 	return conduit, nil
 }
 
-func (n *Network) handleRegisterBlockExchangeRequest(parent irrecoverable.SignalerContext, channel network.Channel, bstore blockstore.Blockstore) (network.BlockExchange, error) {
+func (n *Network) handleRegisterBlobServiceRequest(parent irrecoverable.SignalerContext, channel network.Channel, ds datastore.Batching) (network.BlobService, error) {
 	// TODO: this is a hack, we should not rely on knowing the underlying implementation
 	mw, ok := n.mw.(*Middleware)
 	if !ok {
 		return nil, errors.New("middleware was of unexpected type")
 	}
 	if mw.libP2PNode.dht == nil {
-		return nil, errors.New("block exchange is disabled because content routing is not configured")
+		return nil, errors.New("blob exchange is disabled because content routing is not configured")
 	}
 
-	return NewBlockExchange(parent, mw.libP2PNode.host, mw.libP2PNode.dht, channel.String(), bstore), nil
+	bs := network.NewBlobService(mw.libP2PNode.host, mw.libP2PNode.dht, channel.String(), ds)
+	bs.Start(parent)
+	<-bs.Ready()
```

Comment:
should this check if the context was cancelled?

Reply:
it absolutely should, thanks for catching this

Reply:
Updated so that we skip calling `Ready` altogether. Leave it up to caller to wait for `Ready`.